apps/annotation/opencv_annotation.cpp

```diff
@@ -229,21 +229,29 @@ int main( int argc, const char** argv )
         "{ help h usage ? |      | show this message }"
         "{ images i       |      | (required) path to image folder [example - /data/testimages/] }"
         "{ annotations a  |      | (required) path to annotations txt file [example - /data/annotations.txt] }"
+        "{ maxWindowHeight m  |  -1   | (optional) images larger in height than this value will be scaled down }"
+        "{ resizeFactor r  |  2  | (optional) factor for scaling down [default = half the size] }"
     );
     // Read in the input arguments
     if (parser.has("help")){
         parser.printMessage();
         cerr << "TIP: Use absolute paths to avoid any problems with the software!" << endl;
         return 0;
     }
-    string image_folder(parser.get<string>("images"));
-    string annotations_file(parser.get<string>("annotations"));
+    string image_folder = parser.get<string>("images");
+    string annotations_file = parser.get<string>("annotations");
     if (image_folder.empty() || annotations_file.empty()){
         parser.printMessage();
         cerr << "TIP: Use absolute paths to avoid any problems with the software!" << endl;
         return -1;
     }
 
+    int maxWindowHeight = -1;
+    int resizeFactor = parser.get<int>("resizeFactor");
+    if(parser.get<int>("maxWindowHeight") > 0){
+        maxWindowHeight = parser.get<int>("maxWindowHeight");
+    }
```

Comment:
I was just looking around, and commented when I saw this. Then I looked around some more, and realized there are quite a lot of place in OpenCV that don't mind const-correctness..

I think there's value in being const. suppose you're hunting some bug down.

 * If you see an initialization done right, you see and know it's initialized properly, and you know it will never change, you intuitively conclude it will forever remain right.
 * But if you see something initialized right, and you know it *might* change down the line, then you're still on your guard. You keep reading, looking for the place it changes. And that might not be obvious--for example, some obscure function taking by reference and messing with it, or an assignment from a function that you don't yet know has a bug.

const just means one less moving part. it's just easier to work with..